protected/widgets/Section.php

```diff
@@ -95,6 +107,17 @@ private function initCss(): void
 
     public function run(): string
     {
+        if (isset($this->permission)) {
+            if (!isset($this->subject)) {
+                throw new InvalidConfigException('To use permission, a subject must be set.');
```

Comment:
General: try to avoid nested ifs, instead combine the guards and fail early. Then the happy path is not inside a control statement.

In this case, not having a subject is valid, in which case it would be a global permission. (Same for logic is used for the buttons)